underfs/b2/src/main/java/alluxio/underfs/b2/B2UnderFileSystem.java

```diff
@@ -0,0 +1,374 @@
+import java.io.OutputStream;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * B2 {@link UnderFileSystem} implementation based on the b2-sdk-core library.
+ */
+@ThreadSafe
+public class B2UnderFileSystem extends ObjectUnderFileSystem {
+  private static final Logger LOG = LoggerFactory.getLogger(B2UnderFileSystem.class);
+
+  public static final String B2_SCHEME = "b2://";
+  private static final String FOLDER_SUFFIX = "/";
+
+  private final B2StorageClient mB2StorageClient;
+  private final String mAccountId;
+  private final String mBucketName;
+  private final String mBucketId;
+  private final String mFolderIndicator;
+  private final short mBucketMode;
+
+  protected B2UnderFileSystem(AlluxioURI alluxioURI, UnderFileSystemConfiguration conf,
+      B2StorageClient b2StorageClient, String bucketId, String bucketName, short bucketMode,
+      String accountId) {
+    super(alluxioURI, conf);
+    requireNonNull(alluxioURI, "alluxioURI is null");
+    requireNonNull(conf, "conf is null");
+    requireNonNull(b2StorageClient, "b2StorageClient is null");
+    requireNonNull(bucketId, "bucketId is null");
+    requireNonNull(bucketName, "bucketName is null");
+    requireNonNull(bucketMode, "bucketMode is null");
+    requireNonNull(accountId, "accountId is null");
+
+    LOG.debug("alluxioURI: {}", alluxioURI);
+    LOG.debug("buckedId: {}", bucketId);
+    LOG.debug("bucketName: {}", bucketName);
+    LOG.debug("bucketMode: {}", bucketMode);
+    LOG.debug("accountId: {}", accountId);
+
+    mB2StorageClient = b2StorageClient;
+    mBucketId = bucketId;
+    mBucketName = bucketName;
+    mBucketMode = bucketMode;
+    mAccountId = accountId;
+    mFolderIndicator = conf.get(PropertyKey.B2_FOLDER_INDICATOR);
+  }
+
+  @Override
+  protected boolean copyObject(String src, String dst) throws IOException {
+    LOG.debug("copyObject src: {}, dst: {}", src, dst);
+    // source request
+    B2ListFileVersionsRequest srcRequest =
+        B2ListFileVersionsRequest.builder(mBucketId).setStartFileName(src).setPrefix(src).build();
+    B2FileVersion srcFileVersion = null;
+    try {
+      for (B2FileVersion version : mB2StorageClient.fileVersions(srcRequest)) {
+        if (version.getFileName().equals(src)) {
+          srcFileVersion = version;
+        } else {
+          break;
+        }
+      }
+    } catch (B2Exception e) {
+      e.printStackTrace();
+    }
+    if (srcFileVersion == null) {
+      LOG.error("Source object not found: {}", src);
+      return false;
+    }
+    String sourceFileId = srcFileVersion.getFileId();
+    // destination request
+    B2ListBucketsRequest bucketsRequest =
+        B2ListBucketsRequest.builder(mAccountId).setBucketName(dst).build();
+    try {
+      mB2StorageClient.listBuckets(bucketsRequest);
+    } catch (B2Exception e) {
+      e.printStackTrace();
```

Comment:
Instead of `printStackTrace()`,  can we use the logging framework, to log a message and the exception if we want?

(this comment applies to other locations where `printStackTrace()` is used)